glue/core/data_factories/fits.py

```diff
@@ -114,7 +114,7 @@ def new_data(suffix=True):
             if is_image_hdu(hdu):
                 shape = hdu.data.shape
                 coords = coordinates_from_header(hdu.header)
-                units = hdu.header["BUNIT"] if "BUNIT" in hdu.header else None
+                units = hdu.header.get('BUNIT')if "BUNIT" in hdu.header else None
```

Comment:
Just to clarify, you can just do:

```
units = hdu.header.get('BUNIT')
```

and you don't need the if statement part.